conda-env/dev.yml

```diff
@@ -22,6 +22,7 @@ dependencies:
   - numpy >=2.0.0,<3.0.0
   - pywavelets
   - scipy
+  - uxarray >=2023.3.0,<2025.6.0
```

Comment:
Any reason why we're constraining <2025.6.0 here? Also `ci.yml` doesn't have this constraint.